Improvement: Feedback #3002 (#462)
vtexdocs/dev-portal-content: docs/vtex-io/Storefront-Guides/google-tag-manager/vtex-io-documentation-migrating-google-tag-manager-app.md

---
title: "Migrating Google Tag Manager app from major 2.x to major 3.x"
slug: "vtex-io-documentation-migrating-google-tag-manager-app"
hidden: false
createdAt: "2021-10-27T17:53:23.675Z"
updatedAt: "2023-05-17T09:17:44.057Z"
category: "Storefront Development"
seeAlso:
 - "/docs/guides/vtex-google-tag-manager"
 - "/docs/guides/vtex-io-documentation-setting-up-google-tag-manager"
---

> Version 3.x of VTEX’s Google Tag Manager (GTM) app enables comprehensive tracking of the user's journey within the store, from product views to purchases, using events triggered by the `ecommerceV2` variable. Additionally, it allows stores to send events in both the Google Analytics 4 (GA4) format and the Universal Analytics (UA) format, facilitating a seamless transition to GA4.

Reviewer: ```suggestion
Version 3.x of [VTEX Google Tag Manager (GTM) app](https://developers.vtex.com/docs/apps/vtex.google-tag-manager) introduces enhanced tracking capabilities for user interactions within the store. With the integration of the `ecommerceV2` variable, this version allows for comprehensive monitoring of the user journey, capturing events from product views to completed purchases. Additionally, it allows stores to send events in both the Google Analytics 4 (GA4) format and the Universal Analytics (UA) format, facilitating a seamless transition to GA4.
```

> ## New properties and product schema

Author: ```suggestion
## New properties to the product schema
```

> ## New properties to the product schema

Reviewer: ```suggestion
## Product schema properties of GTM 3.x 
```

To ensure consistency in product information across all areas of the store and effectively track the user's journey, Google Tag Manager 3.x introduces Enhanced Ecommerce properties to the product data schema as events. By adding these properties, stores can include additional information such as product printing, promotion, and sales data, thereby enhancing the overall tracking capabilities and maintaining a unified view of the user's journey throughout the store.

| GTM 3.x property | Type               | Description                                                                 |
| ---------------- | ------------------ | --------------------------------------------------------------------------- |

> | `id`             | Renewed property   | Product ID - Previously SKU ID.                                             |

Reviewer: O que quer dizer com `renewed`?

Author: São props que já existiam, mas foram atualizadas para essa versão. Pensando bem, vou mudar para "updated"

Reviewer: Boa! Pensei em updated mesmo

> | `id`             | Renewed property   | Product ID - Previously SKU ID.                                             |

Author: ```suggestion
| `id`             | Updated property   | Product ID - Previously SKU ID.                                             |
```

> | `variant`        | Renewed property   | SKU ID - Previously SKU Name. The variant of the product, e.g., Rebel pink. |

Author: ```suggestion
| `variant`        | Updated property   | SKU ID - Previously SKU Name. The variant of the product, e.g., Rebel pink. |
```

> | `name`           | Renewed property   | Product Name - Previously Product Name or SKU Name.                         |

Author: ```suggestion
| `name`           | Updated property   | Product Name - Previously Product Name or SKU Name.                         |
```

| `brand`          | Unchanged property | Product brand.                                                              |
| `dimension1`     | New property       | Product Reference ID.                                                       |
| `dimension2`     | New property       | SKU Reference ID.                                                           |
| `dimension3`     | New property       | SKU Name (does not include the Product Name).                               |

The `dimension1`, `dimension2`, `dimension3` properties are custom dimensions that you can use to collect and analyze data that Google Analytics does not automatically create.

For more information about custom dimensions and Enhanced Ecommerce, refer to [Custom dimensions and metrics](https://support.google.com/analytics/answer/2709828?hl=en&ref_topic=2709827#configuration&zippy=%2Cin-this-article) and [Google Enhanced ecommerce official guide](https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-ecommerce#ecommerce-data) respectively.

## Migrating to GTM 3.x

 To migrate to the app's 3.x version, please follow these guides:

> 1. [Create a Google Analytics 4 property](https://developers.vtex.com/docs/guides/vtex-io-documentation-installing-google-tag-manager#create-a-google-analytics-4-property): Refer to the section Create a Google Analytics and follow the instructions provided.

Reviewer: ```suggestion
1. [Create a Google Analytics 4 property](https://developers.vtex.com/docs/guides/vtex-io-documentation-installing-google-tag-manager#create-a-google-analytics-4-property): Refer to the "Create a Google Analytics 4 property" section and follow the instructions provided.
```

> 2. Once you have completed the setup, proceed to the  [Setting up Google Tag Manager documentation](https://developers.vtex.com/docs/guides/vtex-io-documentation-setting-up-google-tag-manager) documentation for instructions on configuring events, tags, and triggers specifically for GA4.

Reviewer: ```suggestion
2. Once you have completed the setup, proceed to the [Setting up Google Tag Manager](https://developers.vtex.com/docs/guides/vtex-io-documentation-setting-up-google-tag-manager) guide for instructions on configuring events, tags, and triggers specifically for GA4.
```
